substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/hub/ClassForNameSupport.java

```diff
@@ -41,12 +41,19 @@ static ClassForNameSupport singleton() {
 
     /** The map used to collect registered classes. */
     private final EconomicMap<String, Class<?>> knownClasses = ImageHeapMap.create();
+    private final EconomicMap<String, String> dynamicGeneratedClasses = ImageHeapMap.create();
```

Comment:
The hash code should be kept in a compact byte array (should be converted during parsing).

Reply:
Fixed by changing `dynamicGeneratedClasses` to EconomicMap<String, byte[]>